core/src/main/java/org/infinispan/factories/GlobalComponentRegistry.java

```diff
@@ -84,6 +85,7 @@
    final Collection<ModuleLifecycle> moduleLifecycles;
 
    final ConcurrentMap<String, ComponentRegistry> namedComponents = new ConcurrentHashMap<String, ComponentRegistry>(4);
+   final ConcurrentMap<ByteString, ComponentRegistry> byteNamedComponents = new ConcurrentHashMap<ByteString, ComponentRegistry>(4);
```

Comment:
you can keep only the ```byteNamedComponents``` map. ```getNamedComponentRegistry(String)``` is only invoked when a cache is started or stopped, so you can afford the cost of converting the ```String``` => ```ByteString``` in that cases.